### PR TITLE
14097 font download android

### PR DESCRIPTION
**Pull request by dominicmarmionswrve**
Added font downloading and using the new cdnPaths.
Changed the Type on Atoms from a String to an ENUM.

https://swrvedev.jira.com/browse/SWRVE-14097

### CODE REVIEW

**Comment by dominicmarmionswrve** [comment]
@seaders @Sergio-Mira @adam-govan please

File: SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/engine/model/ConversationAtom.java [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
 public abstract class ConversationAtom implements Serializable {
-    public static final String TYPE_CONTENT_HTML = "html-fragment";
-    public static final String TYPE_CONTENT_IMAGE = "image";
-    public static final String TYPE_CONTENT_VIDEO = "video";
-    public static final String TYPE_CONTENT_SPACER = "spacer";
-    public static final String TYPE_INPUT_MULTIVALUE = "multi-value-input";
-    public static final String TYPE_INPUT_STARRATING = "star-rating";
+
+    public enum TYPE {
+        @SerializedName("html-fragment")        CONTENT_HTML,
```
Comment (Sergio-Mira) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/engine/model/ConversationAtom.java:12]: 👍

File: SwrveSDK/src/main/java/com/swrve/sdk/conversations/SwrveConversation.java [hunk 2]
```diff
@@ -60,6 +92,20 @@ public boolean areAssetsReady(Set<String> assetsOnDisk) {
         return true;
     }
 
+    private boolean isAssetInCache(Set<String> assetsOnDisk, String asset) {
```
Comment (Sergio-Mira) [inline SwrveSDK/src/main/java/com/swrve/sdk/conversations/SwrveConversation.java]: ~ isImageAsset?

**Comment by Sergio-Mira** [comment]
🚀 (does it do any SHA1 check? note that if we do I've updated the upload to be companyId-SHA1.extension)

**Comment by dominicmarmionswrve** [comment]
The sha is checked here https://github.com/Swrve/swrve-android-sdk/blob/14097_font_download_android/SwrveSDKCommon/src/main/java/com/swrve/sdk/SwrveAssetsManagerImp.java#L123

But it assumes the whole value is encrypted, including the companyid and extension, which would be inconsistent with server. Lets chat about this.....

**Pull request merged**